src/sign_workflow/signer_jar.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from pathlib import Path
+
+from sign_workflow.signer import Signer
+
+"""
+This class is responsible for signing jar and taco files using the OpenSearch-signer-client and verifying its signature.
+"""
+
+
+class SignerJar(Signer):
+    ACCEPTED_FILE_TYPES = [".jar", ".taco"]
+
+    def generate_signature_and_verify(self, artifact: str, basepath: Path, signature_type: str) -> None:
+        filename = os.path.join(basepath, artifact)
+        signed_filename = filename if self.overwrite else os.path.join(basepath, "signed_" + artifact)
+        self.sign(artifact, basepath, signature_type)
+        self.verify(signed_filename)
+
+    def is_valid_file_type(self, file_name: str) -> bool:
+        return any(
+            file_name.endswith(x) for x in SignerJar.ACCEPTED_FILE_TYPES
+        )
+
+    def sign(self, artifact: str, basepath: Path, signature_type: str) -> None:
+        filename = os.path.join(basepath, artifact)
+        signed_filename = filename if self.overwrite else os.path.join(basepath, "signed_" + artifact)
+        signing_cmd = [
+            "./opensearch-signer-client",
+            "-i",
+            filename,
+            "-o",
+            signed_filename,
+            "-p",
+            "jar_signer",
+            "-r",
+            str(self.overwrite)
+        ]
+        self.git_repo.execute(" ".join(signing_cmd))
+
+    def verify(self, filename: str) -> None:
+        verify_cmd = ["jarsigner", "-verify", filename, "-verbose", "-certs", "-strict"]
+        self.git_repo.execute(" ".join(verify_cmd))
```

Comment:
Does this function test the `return_code` after the command has been executed?
You can add a simple check on whether or not `Jar Verified` string exists in the response output or if return_code is not 0 when the jar is not verified successfully.

Reply:
Looks like the case is not same like other signer verification. The jar signer does not exit with non-zero code if jar is not signed. Will add the check based on the text. Thanks!

Reply:
Added this in recent commit. Please review. 
Thanks!